persist/fs/files.go

```diff
@@ -767,3 +830,20 @@ func snapshotPathFromTimeAndIndex(prefix string, t time.Time, suffix string, ind
 	name := fmt.Sprintf("%s%s%d%s%s%s%d%s", filesetFilePrefix, separator, t.UnixNano(), separator, suffix, separator, index, fileSuffix)
 	return path.Join(prefix, name)
 }
+
+func filesetIndexSegmentFilePathFromTime(
+	prefix string,
+	t time.Time,
+	segmentIndex int,
```

Comment:
nit: this is the fileset index, not the segmentIndex, yea?

Reply:
Nah, this is the actual segment number, because you can have multiple segments per index block.  So I need to actually create another one of these functions for the snapshot segment files, right now they just get named without the snapshot index.  I'll add this and add a test for this case too.